src/sentry/api/endpoints/organization_events_spans_performance.py

```diff
@@ -279,6 +282,91 @@ def get_result(self, limit: int, cursor: Optional[Cursor] = None) -> CursorResul
         )
 
 
+class OrganizationEventsSpansStatsEndpoint(OrganizationEventsSpansEndpointBase):
+    def get(self, request: Request, organization: Organization) -> Response:
+        if not self.has_feature(request, organization):
+            return Response(status=404)
+
+        serializer = SpansSerializer(data=request.GET)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+        serialized = serializer.validated_data
+
+        # TODO: The serializer currently supports >1 span.
+        # This will no longer be the case once the examples query is split up.
+        spans = serialized["span"]
+        if len(spans) != 1:
+            raise ParseError(detail="Can only specify 1 span.")
+        span = spans[0]
+
+        def get_event_stats(
+            query_columns: Sequence[str],
+            query: str,
+            params: Dict[str, str],
+            rollup: int,
+            zerofill_results: bool,
+            comparison_delta: Optional[datetime] = None,
+        ) -> SnubaTSResult:
+            with sentry_sdk.start_span(
+                op="discover.discover", description="timeseries.filter_transform"
+            ):
+                builder = TimeseriesQueryBuilder(
+                    Dataset.Discover,
+                    params,
+                    rollup,
+                    query=query,
+                    selected_columns=query_columns,
+                    functions_acl=["array_join", "percentileArray", "sumArray"],
+                )
+
+                spans_op = builder.resolve_function("array_join(spans_op)")
+                spans_group = builder.resolve_function("array_join(spans_group)")
+
+                # Adding spans.op and spans.group to the group by because
+                # We need them in the query to help the array join optimizer
+                # in snuba take effect but the TimeseriesQueryBuilder
+                # removes all non aggregates from the select clause.
+                builder.groupby.extend([spans_op, spans_group])
+
+                builder.add_conditions(
+                    [
+                        Condition(
+                            Function("tuple", [spans_op, spans_group]),
```

Comment:
nit: Mind renaming these variables? This read as a mistake at first (ie. spans_op looks close to span.op)

Reply:
renamed to `span_op_column` and `span_group_column`